tokenizers/javascript/index.js

```diff
@@ -0,0 +1,88 @@
+const esprima = require('esprima')
+const escodegen = require('escodegen')
+const fs = require('fs-extra-promise')
+const tokenizer = require('./tokenizer')
+
+const immutable = require('immutable')
+const walk = require('esprima-walk')
+
+const { base64FileName } = require('./util')
+
+const estools = require('estools')
+
+const TOKENIZER_SCOPE_FILE = 'file-scope'
+const TOKENIZER_SCOPE_FUNCTION = 'function-scope'
+
+const TOKENIZER_SCOPE = TOKENIZER_SCOPE_FILE
+
+// TODO: estools map / filter / traverse (instead of walk)
+// - filter subfunctions from fuction asts somehow
+// - test on SCC
+
+// TODO: get rid of the function block and indentation
+const regenerateFunctionCode = function(functionAst) {
+  codegenOptions = { // NOTE: doesn't help
+    format: {
+      parentheses: false
+    }
+  }
+
+  // NOTE: functionAst.body ommits the function signature (returns block only)
+  return escodegen.generate(functionAst.body, {})
```

Comment:
Yes, other parsers do use method signatures. SourcererCC uses everything in a method except for comments and some symbols. It gives good recall as well as precision with current strategy. May be in future we can setup an experiment to test the hypothesis (clone detection without using method signatures.). But for now, lets use it and make it consistent with other parsers.